runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/MetricsInterceptor.kt

```diff
@@ -28,7 +30,7 @@ internal object MetricsInterceptor : Interceptor {
         }
 
         val originalResponse = chain.proceed(request)
-        val response = if (originalResponse.body.contentLength() != 0L) {
+        val response = if (originalResponse.body?.contentLength() != 0L) {
             originalResponse.newBuilder()
```

Comment:
**Question**: Why is the null-safe operator now necessary? And what does a null `body` imply?

I believe if `body` _is_ actually null then we'll encounter an NPE on L35 where we dereference `body.instrument`...

Reply:
[`Response.body` is nullable in OkHttp3 4.x](https://github.com/square/okhttp/blob/4984568367caaf359b82c452bd28b5e192824d1c/okhttp/src/main/kotlin/okhttp3/Response.kt#L78) but not in 5.x, so I've added null-safe operators everywhere we use a body. 

Good catch that we would still dereference it on L35, I will fix that. In my testing I don't think the body _is_ ever null but it's safer to treat it like so